File: Practica 3/pila.py
class Pila:
    def __init__(self, n):
        self.arreglo = [0] * n
        self.top = -1
        self.n = n

    def push(self, e):
        if not self.isFull():
            self.top += 1
            self.arreglo[self.top] = e

    def pop(self):
        if not self.isEmpty():
            e = self.arreglo[self.top]
            self.top -= 1
            return e

    def peek(self):
        if not self.isEmpty():
            return self.arreglo[self.top]

    def isEmpty(self):
        return self.top == -1

    def isFull(self):
        return self.top == self.n - 1

# Ejemplo de uso
pila = Pila(5)
pila.push(1)
pila.push(2)
print(pila.pop()) # Output: 2
print(pila.peek()) # Output: 1

File: Practica 3/pila.java
public class Pila {
    private long[] arreglo;
    private int top;
    private int n;

    public Pila(int n) {
        this.n = n;
        this.arreglo = new long[n];
        this.top = -1;
    }

    public void push(long e) {
        if (!isFull()) {
            arreglo[++top] = e;
        }
    }

    public long pop() {
        if (!isEmpty()) {
            return arreglo[top--];
        }
        return -1; 
    }

    public long peek() {
        if (!isEmpty()) {
            return arreglo[top];
        }
        return -1; 
    }

    public boolean isEmpty() {
        return top == -1;
    }

    public boolean isFull() {
        return top == n - 1;
    }
}

// Ejemplo de uso
public class Main {
    public static void main(String[] args) {
        Pila pila = new Pila(5);
        pila.push(1);
        pila.push(2);
        System.out.println(pila.pop()); // Output: 2
        System.out.println(pila.peek()); // Output: 1
    }
}
